packages/queued-request-controller/src/QueuedRequestController.ts

```diff
@@ -251,18 +259,25 @@ export class QueuedRequestController extends BaseController<
    */
   async #processNextBatch() {
     const firstRequest = this.#requestQueue.shift() as QueuedRequest;
+    this.#networkClientIdOfCurrentBatch = firstRequest.networkClientId;
     this.#originOfCurrentBatch = firstRequest.origin;
     const batch = [firstRequest.processRequest];
-    while (this.#requestQueue[0]?.origin === this.#originOfCurrentBatch) {
+
+    // alternatively we could still batch by only origin but switch networks in batches by
+    // adding the network clientId to the values in the batch array
```

Comment:
let's just do it this way for now so that we have an MVP. We can improve Approval UX later if we have time

Reply:
agreed! not a big deal for this stepping stone/bandaid approach